From: ohrensessel <git@ohrensessel.net>
Date: Fri, 4 Apr 2014 23:23:43 +0200
Subject: make location and owner info in config mode configurable

disabled by default in hamburg

diff --git a/gluon/gluon-config-mode/files/usr/lib/lua/luci/model/cbi/gluon-config-mode/wizard.lua b/gluon/gluon-config-mode/files/usr/lib/lua/luci/model/cbi/gluon-config-mode/wizard.lua
index de5d42e..59655b7 100644
--- a/gluon/gluon-config-mode/files/usr/lib/lua/luci/model/cbi/gluon-config-mode/wizard.lua
+++ b/gluon/gluon-config-mode/files/usr/lib/lua/luci/model/cbi/gluon-config-mode/wizard.lua
@@ -48,40 +48,44 @@ o.value = uci:get("gluon-simple-tc", meshvpn_name, "limit_egress")
 o.rmempty = false
 o.datatype = "integer"
 
-s = f:section(SimpleSection, nil, [[Um deinen Knoten auf der Karte anzeigen
-zu können, benötigen wir seine Koordinaten. Hier hast du die Möglichkeit,
-diese zu hinterlegen.]])
-
-o = s:option(Flag, "_location", "Knoten auf der Karte anzeigen")
-o.default = uci:get_first("gluon-node-info", "location", "share_location", o.disabled)
-o.rmempty = false
-
-o = s:option(Value, "_latitude", "Breitengrad")
-o.default = uci:get_first("gluon-node-info", "location", "latitude")
-o:depends("_location", "1")
-o.rmempty = false
-o.datatype = "float"
-o.description = "z.B. 53.873621"
+if uci:get_first("gluon-node-info", "location", "disable_location") ~= '1' then
+  s = f:section(SimpleSection, nil, [[Um deinen Knoten auf der Karte anzeigen
+  zu können, benötigen wir seine Koordinaten. Hier hast du die Möglichkeit,
+  diese zu hinterlegen.]])
+
+  o = s:option(Flag, "_location", "Knoten auf der Karte anzeigen")
+  o.default = uci:get_first("gluon-node-info", "location", "share_location", o.disabled)
+  o.rmempty = false
+
+  o = s:option(Value, "_latitude", "Breitengrad")
+  o.default = uci:get_first("gluon-node-info", "location", "latitude")
+  o:depends("_location", "1")
+  o.rmempty = false
+  o.datatype = "float"
+  o.description = "z.B. 53.873621"
+
+  o = s:option(Value, "_longitude", "Längengrad")
+  o.default = uci:get_first("gluon-node-info", "location", "longitude")
+  o:depends("_location", "1")
+  o.rmempty = false
+  o.datatype = "float"
+  o.description = "z.B. 10.689901"
+end
 
-o = s:option(Value, "_longitude", "Längengrad")
-o.default = uci:get_first("gluon-node-info", "location", "longitude")
-o:depends("_location", "1")
-o.rmempty = false
-o.datatype = "float"
-o.description = "z.B. 10.689901"
-
-s = f:section(SimpleSection, nil, [[Hier kannst du einen
-<em>öffentlichen</em> Hinweis hinterlegen um anderen Freifunkern zu
-ermöglichen Kontakt mit dir aufzunehmen. Bitte beachte, dass dieser Hinweis
-auch öffentlich im Internet, zusammen mit den Koordinaten deines Knotens,
-einsehbar sein wird.]])
-
-o = s:option(Value, "_contact", "Kontakt")
-o.default = uci:get_first("gluon-node-info", "owner", "contact", "")
-o.rmempty = true
-o.datatype = "string"
-o.description = "z.B. E-Mail oder Telefonnummer"
-o.maxlen = 140
+if uci:get_first("gluon-node-info", "owner", "disable_owner") ~= '1' then
+  s = f:section(SimpleSection, nil, [[Hier kannst du einen
+  <em>öffentlichen</em> Hinweis hinterlegen um anderen Freifunkern zu
+  ermöglichen Kontakt mit dir aufzunehmen. Bitte beachte, dass dieser Hinweis
+  auch öffentlich im Internet, zusammen mit den Koordinaten deines Knotens,
+  einsehbar sein wird.]])
+
+  o = s:option(Value, "_contact", "Kontakt")
+  o.default = uci:get_first("gluon-node-info", "owner", "contact", "")
+  o.rmempty = true
+  o.datatype = "string"
+  o.description = "z.B. E-Mail oder Telefonnummer"
+  o.maxlen = 140
+end
 
 function f.handle(self, state, data)
   if state == FORM_VALID then
@@ -116,16 +120,20 @@ function f.handle(self, state, data)
     uci:save("system")
     uci:commit("system")
 
-    local sname = uci:get_first("gluon-node-info", "location")
-    uci:set("gluon-node-info", sname, "share_location", data._location)
-    if data._location and data._latitude ~= nil and data._longitude ~= nil then
-      uci:set("gluon-node-info", sname, "latitude", data._latitude)
-      uci:set("gluon-node-info", sname, "longitude", data._longitude)
+    if uci:get_first("gluon-node-info", "location", "disable_location") ~= '1' then
+      local sname = uci:get_first("gluon-node-info", "location")
+      uci:set("gluon-node-info", sname, "share_location", data._location)
+      if data._location and data._latitude ~= nil and data._longitude ~= nil then
+        uci:set("gluon-node-info", sname, "latitude", data._latitude)
+        uci:set("gluon-node-info", sname, "longitude", data._longitude)
+      end
     end
-    if data._contact ~= nil then
-      uci:set("gluon-node-info", uci:get_first("gluon-node-info", "owner"), "contact", data._contact)
-    else
-      uci:delete("gluon-node-info", uci:get_first("gluon-node-info", "owner"), "contact")
+    if uci:get_first("gluon-node-info", "owner", "disable_owner") ~= '1' then
+      if data._contact ~= nil then
+        uci:set("gluon-node-info", uci:get_first("gluon-node-info", "owner"), "contact", data._contact)
+      else
+        uci:set("gluon-node-info", uci:get_first("gluon-node-info", "owner"), "contact")
+      end
     end
     uci:save("gluon-node-info")
     uci:commit("gluon-node-info")
diff --git a/gluon/gluon-node-info/files/etc/config/gluon-node-info b/gluon/gluon-node-info/files/etc/config/gluon-node-info
index 9ac6ec5..cf3a16a 100644
--- a/gluon/gluon-node-info/files/etc/config/gluon-node-info
+++ b/gluon/gluon-node-info/files/etc/config/gluon-node-info
@@ -1,4 +1,6 @@
 config location
 	option share_location '0'
+	option disable_location '1'
 
 config owner
+	option disable_owner '1'
